### PR TITLE
Adding an operator-upgrade Github workflow.

### PR DESCRIPTION
**Pull request by ybettan**
This test will ensure the bundle built in the PR can be used to upgrade from the last bundle available of KMM.

### CODE REVIEW

**Comment by ybettan** [comment]
/cc @qbarrand 
It has passed 5/5 of the tries I did.

I have kept the explicit namespaces collection since although the default behavior [documented](https://troubleshoot.sh/docs/collect/cluster-resources/#namespaces-optional) is to collect data from all namespace I still got partial data and some of the objects were empty.

File: ci/prow/operator-upgrade [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# Install OLM
+make operator-sdk
+./bin/operator-sdk olm install
+
+# It takes some time to the operatorhubio-catalog catalogsource to get to a
+# READY connection state, we need this to validate that the catalog is ready
+# before installing a bundle, otherwise we might get a race condition that will
+# resolved in the Subscription not being reconciled after the connecting is
+# ready
+timeout 3m bash -c 'until kubectl -n olm get catalogsource/operatorhubio-catalog -o yaml | yq ".status.connectionState.lastObservedState == \"READY\"" | grep true; do sleep 5; done'
```
Comment (qbarrand) [inline ci/prow/operator-upgrade:31]:
```suggestion
timeout 3m bash -c 'until [ "$(kubectl -n olm get catalogsource/operatorhubio-catalog -o jsonpath={.status.connectionState.lastObservedState})" = "READY" ]; do sleep 5; done'
```

I think this is slightly cleaner. Then the rest lgtm.

**Pull request merged**